src/components/app-configure/access/DidDelegate.vue

```diff
@@ -20,14 +24,72 @@ const loaderStore = useLoaderStore()
 const toast = useToast()
 
 const app = appsStore.app(appId)
-const delegates = computed(() => app.access.delegates)
+const delegates = ref(app.access.delegates)
 
-const showDeletePopup = ref(false)
+type selectedOperation = 'create' | 'edit'
 
+const generatedKeyInfo = ref({ address: '', privateKey: '' })
+const delegateKeys: Ref<{ name: string; address: string }[]> = ref([])
+const showCreateDelegate = ref(false)
+const showGenerateKeySuccess = ref(false)
+const showEditDelegate = ref(false)
+const showDeletePopup = ref(false)
+const delegateToEdit = ref({})
 const selectedDelegate: Ref<Delegate | null> = ref(null)
+let currentSelectedOperation: selectedOperation = 'edit'
+
+async function getDelegateKeys() {
+  try {
+    loaderStore.showLoader('Fethcing keys...')
+    const { data } = await listDelegateKeys(app.address)
+    delegateKeys.value = data
+  } catch (err) {
+    console.log({ err })
+  } finally {
+    loaderStore.hideLoader()
+  }
+}
 
 async function addDelegate() {
-  //
+  currentSelectedOperation = 'create'
+  await getDelegateKeys()
+  showCreateDelegate.value = true
+}
+
+function onGenerateClick() {
+  const { address, privateKey } = generateKey()
+  generatedKeyInfo.value.address = address
+  generatedKeyInfo.value.privateKey = privateKey
+  delegateKeys.value = [...delegateKeys.value, { name: address, address }]
+  showCreateDelegate.value = false
+  showEditDelegate.value = false
+  showGenerateKeySuccess.value = true
+}
+
+async function onCreatingDelegate() {
+  await appsStore.fetchAndStoreAppConfig(appId)
+  const app = appsStore.app(appId)
+  delegates.value = app.access.delegates
+}
+
+async function onEditingDelegate() {
+  await appsStore.fetchAndStoreAppConfig(appId)
+  const app = appsStore.app(appId)
+  delegates.value = app.access.delegates
+}
+
+async function onEditClick(delegate: object) {
```

Comment:
Can we use a more specific type for `delegate`?

Reply:
https://github.com/arcana-network/developer-dashboard/pull/156/commits/04175c6cb8d355df114957bc9ef3b852faae92f2